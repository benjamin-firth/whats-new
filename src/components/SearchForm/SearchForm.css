.searchForm {
  align-items: center;
  background-color: #DCD9BA;
  display: flex;
  height: 10%;
  width: 100%;
}
