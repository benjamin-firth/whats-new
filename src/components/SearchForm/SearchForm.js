import React from 'react';
import './SearchForm.css';

const SearchForm = () => {
  return (
    <header className='searchForm'>
      <h1>What's New!?</h1>
      <input type='text' placeholder='Search Articles by Keyword'/>
      <button className='searchButton'>Enter</button>
    </header>
  )
}
export default SearchForm;