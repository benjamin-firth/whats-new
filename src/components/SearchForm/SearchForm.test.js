import React from 'react';
import { shallow  } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let wrapper;
  let mockSearchHandler;
  
  beforeEach(() => {
    mockSearchHandler = jest.fn();
    wrapper = shallow(<SearchForm 
      searchHandler={mockSearchHandler}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is invoked', () => {
    const mockEvent = {target: {value: 'test'}};
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('keyword')).toEqual('test');
  })

  it('should call the the searchHandler method when button is clicked', () => {

    wrapper.find('button').simulate('click');
    expect(mockSearchHandler).toHaveBeenCalled();
  })
})