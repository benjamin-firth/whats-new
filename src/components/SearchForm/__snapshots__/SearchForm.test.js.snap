// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<header
  className="searchForm"
>
  <h1>
    What's
    <span
      className="boldTitle"
    >
      New?
    </span>
  </h1>
  <form>
    <input
      id="search"
      onChange={[Function]}
      placeholder="Filter Articles by Keyword"
      type="text"
    />
    <button
      className="searchButton"
      onClick={[Function]}
      type="button"
    >
      Enter
    </button>
  </form>
</header>
`;
